src/Ledger/Rewards.lagda

```diff
@@ -0,0 +1,172 @@
+\section{Rewards}
+\label{sec:rewards}
+\modulenote{\LedgerModule{Rewards}}
+
+\begin{code}[hide]
+{-# OPTIONS --safe #-}
+
+open import Agda.Builtin.FromNat
+
+open import Data.Rational using (ℚ; floor; _*_; _÷_; _/_; _-_)
+import Data.Rational as ℚ renaming (_⊓_ to min; _⊔_ to max)
+open import Data.Rational.Literals using (number)
+import Data.Rational.Properties as ℚ
+import Data.Nat as ℕ renaming (_⊔_ to max)
+import Data.Integer as ℤ renaming (_⊔_ to max)
+import Data.Integer.Properties as ℤ
+open Number number renaming (fromNat to fromℕ)
+
+open import Ledger.Prelude hiding (_/_; _*_; _-_)
+open import Ledger.Types.GovStructure
+open import Ledger.Types.Numeric.UnitInterval
+
+module Ledger.Rewards
+  (gs : _) (open GovStructure gs)
+  where
+
+open import Ledger.Certs gs
+\end{code}
+This section defines how rewards for stake pools and their delegators
+are calculated and paid out.
+
+\subsection{Rewards Motivation}
+\label{sec:rewards-motivation}
+In order to operate, any blockchain needs to attract parties that are
+willing to spend computational and network resources
+on receiving transactions and producing new blocks.
+These parties, called \defn{block producers},
+are incentivized by monetary \defn{rewards}.
+
+Cardano is a proof of stake (PoS) blockchain:
+Through a random lottery,
+one block producer is selected to produce one particular block.
+The probability for being select depends on their \defn{stake} of Ada,
+that is their attributed amount of Ada relative to the total amount of Ada.
+After successful block production,
+the block producer is eligible for a share of the rewards.
+
+The rewards for block producers come from two sources:
+During an initial period, rewards are paid out from the \defn{reserve},
+which is an initial allocation of Ada created for this very purpose.
+Over time, the reserve is depleted,
+and rewards are sourced from transaction fees.
+
+Rewards are paid out epoch by epoch.
+
+Rewards are collective, but depend on performance:
+After every epoch, a fraction of the available reserve
+and the transaction fees accumulated during that epoch
+are added together. This sum is paid out to the block producers
+proportionally to how many blocks they have created each.
+In order to avoid perverse incentives, block producers
+do not receive individual rewards that depend on the content
+of their blocks.
+
+Not all people can or want to set up and administier a dedicated computer
+that produces blocks. However, these people still own Ada,
+and their stake is relevant for block production.
+Specifically, these people have the option to \defn{delegate} their stake
+to a \defn{stake pool}, which belongs to a block producer.
+The rewards of the block producer are shared proportionally with the
+people that delegate to its stake pool, after fees.
```

Comment:
```suggestion
The block producer shares the rewards proportionally with the people that delegated their Ada to its stake pool, after fees.
```
I find the sentence hard to understand. I rewrote it in active voice, maybe it helps.